kong/db/schema/entities/upstreams.lua

```diff
@@ -186,6 +190,10 @@ local r =  {
     { hash_fallback_header = typedefs.header_name, },
     { hash_on_cookie = { type = "string",  custom_validator = utils.validate_cookie_name }, },
     { hash_on_cookie_path = typedefs.path{ default = "/", }, },
+    { hash_on_query_arg = simple_param },
+    { hash_fallback_query_arg = simple_param },
+    { hash_on_uri_capture = simple_param },
+    { hash_fallback_uri_capture = simple_param },
```

Comment:
I think we will need compatibility version fix for this, is it exported to DPs ?

Reply:
Hmm, probably do need some logic to clear these out, but only if they are `null` (otherwise we're just shipping a broken upstream to the DP).

Reply:
if they have a value, the entity will break if sent to old DPs due to the unknown field, right?

Reply:
Right, if they have a value then that means that `hash_on` is probably set to one of the new values (`path`, `query_arg`, `uri_capture`), in which case there's not really any way to make the upstream compatible with the DP.

Reply:
I see, okay